crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -1109,7 +1116,28 @@ pub(crate) async fn get_tx_status(
         None => {
             let status = tx_status_manager.status(id).await?;
             match status {
-                Some(status) => Ok(Some(TransactionStatus::new(id, status))),
+                Some(status) => {
```

Comment:
<img width="746" alt="image" src="https://github.com/user-attachments/assets/d1a7eaf8-c92f-404a-bbaf-3339bb5103f3" />


`StatusChangeState` also will benefit from this change, because we will not lose `Submitted` status. It owed be nice to use this function there, because it works with exact same types=)

Reply:
The database types doesn't seems to be the same and the tx_status_manager neither. I tried different things but couldn't get a compiling version. I don't want to spend too much time on that if you have the idea to do it it will be more efficient.